React/main/src/redux/steps/reducer.js

```diff
@@ -1,15 +1,13 @@
+import { createReducer } from 'redux-recompose';
+
 const initialState = {
   stepNumber: 0
 };
 
-export function steps(state = initialState, action) {
-  switch (action.type) {
-    case 'ADD_STEP':
-      return {
-        ...state,
-        stepNumber: action.step
-      };
-    default:
-      return state;
-  }
-}
+const addStep = (state, action) => ({ ...state, [action.target]: action.payload });
```

Comment:
We recommend change it to state.merge({ [action.target]: action.payload }) because we want an inmutable state